packages/worker/src/worker.ts

```diff
@@ -1746,6 +1742,14 @@ async function extractActivityInfo(
   const { taskToken } = task as NonNullableObject<coresdk.activity_task.IActivityTask>;
   const start = task.start as NonNullableObject<coresdk.activity_task.IStart>;
   const activityId = start.activityId;
+  let heartbeatDetails = undefined;
+  try {
+    heartbeatDetails = await decodeFromPayloadsAtIndex(dataConverter, 0, start.heartbeatDetails);
+  } catch (e) {
+    throw ApplicationFailure.fromError(e, {
+      message: `Failed to parse heartbeat details for activity ${activityId}: ${errorMessage(e)}`,
+    });
+  }
   return {
```

Comment:
Can you make sure that if this code throws we log and error and dump the task we got so we can inspect this better?

Reply:
I added log, including `proto.toJSON()` and the unaltered protbuf-encoded buffer (in base64 notation).

```
2024-07-24T04:45:57.463Z [ERROR] Error while processing ActivityTask.start: Expected scheduledTime to be a timestamp, got null {
  sdkComponent: 'worker',
  taskQueue: 'test',
  error: TypeError: Expected scheduledTime to be a timestamp, got null
      at requiredTsToMs (/Users/jwatkins/Development/Temporal/TypeScriptSDK/sdk-typescript-2/packages/common/src/time.ts:37:11)
      at Worker.extractActivityInfo (/Users/jwatkins/Development/Temporal/TypeScriptSDK/sdk-typescript-2/packages/worker/src/worker.ts:1064:43)
      at runNextTicks (node:internal/process/task_queues:60:5)
      at listOnTimeout (node:internal/timers:540:9)
      at process.processTimers (node:internal/timers:514:7)
      at <anonymous> (/Users/jwatkins/Development/Temporal/TypeScriptSDK/sdk-typescript-2/packages/worker/src/worker.ts:870:28),
  task: '{"taskToken":"ODQxYjEyM2EtYjc5MS00ZmE3LThmODctMTI3YjgxOWZlZWUw","start":{"workflowExecution":{"workflowId":"wfid","runId":"runId"},"activityType":"dummy","currentAttemptScheduledTime":{"seconds":"0","nanos":0},"startedTime":{"seconds":"0","nanos":0},"attempt":1,"scheduleToCloseTimeout":{"seconds":"60","nanos":0},"startToCloseTimeout":{"seconds":"60","nanos":0},"heartbeatTimeout":{"seconds":"60","nanos":0}}}',
  taskEncoded: 'CiQ4NDFiMTIzYS1iNzkxLTRmYTctOGY4Ny0xMjdiODE5ZmVlZTAaNhoNCgR3ZmlkEgVydW5JZCoFZHVtbXlSBAgAEABaBAgAEABgAWoECDwQAHIECDwQAHoECDwQAA=='
}
```